src/Spline.cpp

```diff
@@ -216,7 +231,7 @@ double SplineBase::getCurvatureMax()
     if (has_curvature_max)
         return curvature_max;
 
-    double const delPara = getUnitParameter() * geometric_resolution;
+    double const delPara = geometric_resolution;
```

Comment:
This change makes no sense. The search is in parameter space, geometric_resolution in geometrical space. If you would like to search in geometrical space, you would need to do a binary search to "advance" by geometric_resolution each point.

I seem to remember there is a way to (1) get the derivative curve of the spline and (2) get the max of a curve. That would also be a much better option.